package trove.util.io

/*
 *  # Trove
 *
 *  This file is part of Trove - A FREE desktop budgeting application that
 *  helps you track your finances, FREES you from complex budgeting, and
 *  enables you to build your TROVE of savings!
 *
 *  Copyright Â© 2016-2019 Eric John Fredericks.
 *
 *  Trove is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Trove is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Trove.  If not, see <http://www.gnu.org/licenses/>.
 */

// ejf-fixMe: implement
class IOUsingSpec {
  // "using" should "return the value produced by the function passed to it"
  // it should "call close if the function succeeds"
  // it should "call close if the function succeeds and throw any exception thrown by close"
  // it should "call close if the function throws and throw the original exception if close doesn't throw"
  // it should "call close if the function throws and throw the original exception suppressing a non-fatal exception thrown by close"
  // it should "call close if the function throws and throw a fatal exception thrown by close suppressing an original non-fatal exception thrown by the function"
  // it should "call close if the function throws and throw an interrupted exception thrown by close suppressing a non-fatal exception thrown by the function"
  // it should "call close if the function throws and throw an interrupted exception thrown by close suppressing a fatal exception thrown by the function"
  // it should "call close if the function throws and throw an original fatal exception thrown by the function suppressing a fatal exception that is not an interrupted exception thrown by close"

}
